### ISSUE TITLE
`flutter drive` does output `print`, swallows exception info when test fails

### ISSUE THREAD

**Issue by yjbanov** [flutter/flutter#69070]
## Steps to Reproduce

1. In the Flutter repo `cd dev/integration_tests/flutter_gallery`
2. Edit `dev/integration_tests/flutter_gallery/test_driver/run_demos.dart` and throw an exception in the middle of the test, e.g. right after one of `pumpAndSettle()`. Also add some `print` statements prior to throwing the exception.
3. Run this command: `flutter drive --driver=test_driver/transitions_perf_e2e_test.dart --target=test_driver/transitions_perf_e2e.dart --browser-name=chrome -d web-server --profile`

**Expected results:**

The test prints the output from the `print` statements, then fails outputting the exception message and stack trace.

**Actual results:**

All you see in the output is this:

```
$ flutter drive --driver=test_driver/transitions_perf_e2e_test.dart --target=test_driver/transitions_perf_e2e.dart --browser-name=chrome -d web-server --profile
Running "flutter pub get" in flutter_gallery...                    450ms
Launching test_driver/transitions_perf_e2e.dart on Web Server in profile mode...
Compiling test_driver/transitions_perf_e2e.dart for the Web...        30.5s
test_driver/transitions_perf_e2e.dart is being served at http://localhost:46067
The web-server device requires the Dart Debug Chrome extension for debugging. Consider using the Chrome or Edge devices for an improved development
workflow.
Application finished.
Failure Details:
Failure in method: all demos

end of failure 1
```

This is reproducible in e7b66ac7ad45728d19e67034255fb9c303430c0e

**Comment by nturgut** [comment]
To my knowledge this only happens on profile mode. In debug mode you can see the stack trace, therefore I changed the title to profile mode.

I usually run on debug mode to debug/troubleshoot a problem. Do you think since one mode is showing the error, this is still P3? @yjbanov

**Comment by nturgut** [comment]
Btw, the error in this case is not on e2e, it's on flutter_test. TextExceptionReporter doesn't receive an error message on profile/release modes: https://github.com/flutter/plugins/blob/master/packages/integration_test/lib/integration_test.dart#L69

Therefore this does not need a code change on e2e but on flutter/flutter repo. 

Another information is as you would remember this used to work on March (We specifically tested this [PR](https://github.com/flutter/plugins/pull/2593) on profile mode back in the day.) after a while, TextExceptionReporter stopped working for web profile mode. We might want to add a test to flutter/flutter

@yjbanov @dnfield

**Comment by yjbanov** [comment]
I agree, I think the issue is in `flutter drive`/`package:flutter_driver` more than in `package:integration_test`. When running a test we should always print the output of both the driver script and browser logs, whether we use `package:integration_test` or not.

I'm currently trying to solve it by configuring WebDriver to capture the browser output. I'm getting some partial output, but not all of it. Or at least so it seems. It could also be that the test I'm running abruptly ends and doesn't produce any output, in which case it could be more than one issue. I'm still investigating and will keep in touch.

**Comment by yjbanov** [comment]
Punting on this for now. I spend an hour and wasn't able to figure this out. This needs more investigation.

**Comment by Hixie** [comment]
Do we have a minimal reproduction for this? I'm not quite sure I follow the sequence of steps that's needed to cause this failure.

**Comment by golane-august** [comment]
For me it doesn't even properly run on "debug" mode. 

And here it is stated: `Debug mode will be supported soon.` What does that mean @yjbanov ?
https://github.com/flutter/flutter/wiki/Running-Flutter-Driver-tests-with-Web#running-flutter-driver-tests

Running with `-d chrome` works fine, getting all debug messages. Also Chrome `99+` added even more problems. `<=98` at least is running the tests.

**Comment by kenzieschmoll** [comment]
Any updates on this issue? We are also running into this for writing integration tests for DevTools

**Comment by kenzieschmoll** [comment]
This is preventing us from being able to run our integration tests in profile mode on our CI. We use the stdout and stderr output to detect errors or exceptions thrown from any part of our test script.

**Comment by desmond206x** [comment]
+1 Any news?
